src/interpreter/inner.rs

```diff
@@ -77,16 +81,20 @@ pub enum ScriptType {
     Sh,
     Wsh,
     ShWsh,
+    #[allow(dead_code)]
+    Tr, // Script Spend
 }
 
 /// Structure representing a script under evaluation as a Miniscript
 #[derive(Clone, PartialEq, Eq, Debug)]
-pub enum Inner {
+pub(super) enum Inner {
     /// The script being evaluated is a simple public key check (pay-to-pk,
     /// pay-to-pkhash or pay-to-witness-pkhash)
-    PublicKey(bitcoin::PublicKey, PubkeyType),
+    // Technically, this allows representing a (XonlyKey, Sh) output but we make sure
+    // that only appropriate outputs are created
+    PublicKey(super::BitcoinKey, PubkeyType),
     /// The script being evaluated is an actual script
-    Script(Miniscript<bitcoin::PublicKey, NoChecksEcdsa>, ScriptType),
+    Script(Miniscript<super::BitcoinKey, NoChecksEcdsa>, ScriptType),
```

Comment:
When would we want a TR with an ECDSA here?

Reply:
We never want it. But we have to be careful here to not ever create it.